File: android-client/app/src/main/java/com/example/familytreeplatform/data/local/PersonEntity.kt
package com.example.familytreeplatform.data.local

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.example.familytreeplatform.models.PersonListItem
import com.example.familytreeplatform.models.PersonResponse

@Entity(tableName = "persons", indices = [Index("spaceId")])
data class PersonEntity(
    @PrimaryKey val personId: String,
    val spaceId: String,
    val fullName: String,
    val createdAt: String,
    val lifeStatus: String,
    val deceasedAt: String?,
    val birthDate: String?,
    val birthPlace: String? = null,
    val gender: String?,
    val notes: String? = null,
    val nickName: String? = null,
    val deathPlace: String? = null,
    @ColumnInfo(defaultValue = "1") val version: Int = 1,
    @ColumnInfo(defaultValue = "'FAMILY'") val visibility: String = "FAMILY",
    @ColumnInfo(defaultValue = "'FULL'") val privacyAccess: String = "FULL",
    @ColumnInfo(defaultValue = "0") val canManageVisibility: Boolean = false
)

fun PersonEntity.toModel() = PersonListItem(
    personId = personId,
    fullName = fullName,
    createdAt = createdAt,
    lifeStatus = lifeStatus,
    deceasedAt = deceasedAt,
    birthDate = birthDate,
    birthPlace = birthPlace,
    gender = gender,
    notes = notes,
    version = version,
    nickName = nickName,
    deathPlace = deathPlace,
    visibility = visibility,
    privacyAccess = privacyAccess,
    canManageVisibility = canManageVisibility
)

fun PersonListItem.toEntity(spaceId: String) = PersonEntity(
    personId = personId,
    spaceId = spaceId,
    fullName = fullName,
    createdAt = createdAt,
    lifeStatus = lifeStatus,
    deceasedAt = deceasedAt,
    birthDate = birthDate,
    birthPlace = birthPlace,
    gender = gender,
    notes = notes,
    nickName = nickName,
    deathPlace = deathPlace,
    version = version,
    visibility = visibility,
    privacyAccess = privacyAccess,
    canManageVisibility = canManageVisibility
)

fun PersonResponse.toEntity() = PersonEntity(
    personId = personId,
    spaceId = spaceId,
    fullName = fullName,
    createdAt = createdAt,
    lifeStatus = lifeStatus ?: "ALIVE",
    deceasedAt = deceasedAt,
    birthDate = birthDate,
    birthPlace = birthPlace,
    gender = gender,
    notes = notes,
    nickName = nickName,
    deathPlace = deathPlace,
    version = version,
    visibility = visibility,
    privacyAccess = privacyAccess,
    canManageVisibility = canManageVisibility
)

File: backend/src/persons/person.entity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
  VersionColumn,
} from 'typeorm';

@Entity('persons')
export class PersonEntity {
  @PrimaryGeneratedColumn('uuid')
  personId!: string;

  @Column({ type: 'uuid' })
  spaceId!: string; // FamilySpace

  @Column({ type: 'text' })
  fullName!: string;

  @Column({ type: 'text', nullable: true })
  title!: string | null;

  @Column({ type: 'text', nullable: true })
  firstName!: string | null;

  @Column({ type: 'text', nullable: true })
  lastName!: string | null;

  @Column({ type: 'text', nullable: true })
  suffix!: string | null;

  @Column({ type: 'text', nullable: true })
  nickName!: string | null;

  @Column({ type: 'text', nullable: true })
  gender!: string | null;

  @Column({ type: 'date', nullable: true })
  birthDate!: string | null;

  @Column({ type: 'text', nullable: true })
  birthPlace!: string | null;

  @Column({ type: 'date', nullable: true })
  deathDate!: string | null;

  @Column({ type: 'text', nullable: true })
  deathPlace!: string | null;

  @Column({ type: 'text', nullable: true })
  idNumber!: string | null;

  @Column({ type: 'text', nullable: true })
  notes!: string | null;

  @Column({ type: 'text', default: 'ALIVE' })
  lifeStatus!: 'ALIVE' | 'DECEASED' | 'UNKNOWN';

  @Column({ type: 'date', nullable: true })
  deceasedAt!: string | null;

  @VersionColumn({ type: 'int', default: 1 })
  version!: number;

  @Column({ type: 'boolean', default: false })
  isDeleted!: boolean;

  @Column({ name: 'deleted_at', type: Date, nullable: true })
  deletedAt!: Date | null;

  @CreateDateColumn()
  createdAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;
}
